client/db/types.go

```diff
@@ -184,7 +184,19 @@ type AccountInfo struct {
 	LegacyFeeCoin    []byte
 	LegacyFeeAssetID uint32
 	// LegacyFeePaid should be set on retrieval if there is an AccountProof set.
-	LegacyFeePaid bool // DEPRECATED
+	LegacyFeePaid bool // DEPRECATED LegacyFeeConfirmed
+}
+
+// Registered indicates if the account should exist server-side. This is when
+// the legacy fee txn or at least one bond txn has been broadcast even if
+// neither has received the required confirmations. This may return a false
+// negative for a recovered account that had no active bonds at recovery time,
+// unless new bond(s) have been posted since recovery.
+func (ai *AccountInfo) Registered() bool {
+	if ai.LegacyFeePaid || len(ai.LegacyFeeCoin) > 0 {
+		return true
+	}
+	return len(ai.Bonds) > 0
```

Comment:
I see. Yeah, there's a need to discern if a bond tx or the legacy fee tx has been broadcast to prevent re-attempting a new legacy tx. But that's about it, and since legacy registration will soon be a thing of the past, I'm beginning to wonder if there's any need to account for this.

The only thing we'd need to know is if there's NO active/pending bond and then display a prompt on the UI to post a bond to trade. If there's a pending bond, we display a confirmations counter.

Reply:
Alright, sounds fine for now.  Will test it all out today.